src/core/services/layout/layout.js

```diff
@@ -1,319 +1,441 @@
+
+    .directive('hideLtMd', warnAttrNotSupported('hide-lt-md'))
+    .directive('hideLtLg', warnAttrNotSupported('hide-lt-lg'))
+    .directive('showLtMd', warnAttrNotSupported('show-lt-md'))
+    .directive('showLtLg', warnAttrNotSupported('show-lt-lg'));
+
+  /**
+   * These functions create registration functions for ngMaterial Layout attribute directives
+   * This provides easy translation to switch ngMaterial attribute selectors to
+   * CLASS selectors and directives; which has huge performance implications
+   * for IE Browsers
+   */
+
+  /**
+   * Creates a directive registration function where a possbile dynamic attribute value will
+   * be observed/watched.
+   * @param {string} className attribute name; eg `layout-gt-md` with value ="row"
+   */
+  function attributeWithObserve(className) {
+
+    return ['$mdUtil', '$$mdLayout', '$document', '$parse', '$interpolate', function(_$mdUtil_, _$$mdLayout_, $document, _$parse_, _$interpolate_) {
+      $mdUtil = _$mdUtil_;
+      $parse = _$parse_;
+      $$mdLayout = _$$mdLayout_;
+      $interpolate = _$interpolate_;
+
+      return {
+        restrict: 'A',
+        compile: function(element, attr) {
+          var linkFn = angular.noop;
+
+          // Use for postLink to account for transforms after ng-transclude.
+          if (config.enabled) {
+
+            // immediately replace static (non-interpolated) invalid values...
+            validateAttributeValue( className,
+              getNormalizedAttrValue(className, attr, ""),
+              buildUpdateFn(element, className, attr)
+            );
+
+            linkFn = translateToClass_attributeWithValue;
+          }
 
-    /**
-     * These functions create registration functions for ngMaterial Layout attribute directives
-     * This provides easy translation to switch ngMaterial attribute selectors to
-     * CLASS selectors and directives; which has huge performance implications
-     * for IE Browsers
-     */
+          return linkFn;
+        }
+      };
+    }];
 
     /**
-     * Creates a directive registration function where a possbile dynamic attribute value will
-     * be observed/watched.
-     * @param {string} className attribute name; eg `md-layout-gt-md` with value ="row"
-     */
-    function attributeWithObserve(className) {
-
-      return ['$mdUtil', '$$mdLayout', '$document', '$parse', '$interpolate', function(_$mdUtil_, _$$mdLayout_, $document, _$parse_, _$interpolate_) {
-        $mdUtil = _$mdUtil_;
-        $$mdLayout = _$$mdLayout_;
-        $parse = _$parse_;
-        $interpolate = _$interpolate_;
-
-        return {
-            restrict : 'A',
-            compile: function(element, attr) {
-              // Use for postLink to account for transforms after ng-transclude.
+      * Add as transformed class selector(s), then
+      * remove the deprecated attribute selector
+      */
+     function translateToClass_attributeWithValue(scope, element, attrs) {
```

Comment:
The underscores are an internal AngularJS trick to have the same named argument injected as a function parameter.
I guess in this case (since I annotated, I do not need this... but it is a known, accepted standard)